sdlib/d/gc/tcache.d

```diff
@@ -88,11 +89,20 @@ public:
 		}
 
 		auto pd = getPageDescriptor(ptr);
-
 		auto e = pd.extent;
-		// Slab is not yet supported
-		if (e !is null && !pd.isSlab() && e.finalizer !is null) {
-			e.finalizer(ptr, e.usedCapacity);
+
+		if (e !is null) {
```

Comment:
e being null is a failure here, no? Even an interior pointer is. At least it is for free, and it'll be when calling arena's free.

Reply:
In fact, if ptr isn't the base, the finalizer call is screwed up.

Reply:
That does not check that the pointer isn't an interior pointer.